mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -322,7 +324,11 @@ object PowerIterationClustering extends Logging {
       }, preservesPartitioning = true).cache()
     val sum = r.values.map(math.abs).sum()
     val v0 = r.mapValues(x => x / sum)
-    Graph(VertexRDD(v0), g.edges)
+    val graph = Graph(VertexRDD(v0), g.edges)
+    materialize(graph)
```

Comment:
`materialize` runs `foreachPartition` on the graph's edges. I didn't see it is persisted, so not sure why you want to materialize it without caching it?

Reply:
Same deal, it's really materializing the edges and vertices RDDs.